grails-app/services/org/pih/warehouse/core/DashboardService.groovy

```diff
@@ -174,10 +188,25 @@ class DashboardService {
         }
 
         if (expirationStatus) {
+            def daysToExpiry = [30, 60, 90, 180, 365]
+
             expiringStock = expiringStock.findAll { inventoryItem ->
-                (inventoryItem.expirationStatus == expirationStatus)
+                if (expirationStatus == "greaterThan365Days") {
+                    inventoryItem.expirationDate - today > 365
+                } else if (expirationStatus == "expired") {
+                    inventoryItem.expirationDate - today <= 0
+                } else inventoryItem
+            }
+
+            daysToExpiry.each { day ->
+                if (expirationStatus == "within${day}Days") {
+                    expiringStock = expiringStock.findAll { inventoryItem ->
```

Comment:
I can't tell what's going on here. Wouldn't this assignment override the earlier assignment?

Reply:
Nevermind, I see it now. This seems like a long way around the problem, but I guess you're trying to avoid a long branching statement. Instead of iterating over daysToExpiry, you could just pattern match the expirationStatus string, extrapolate the `days` value and go from there. But whatevs.

Reply:
@jmiranda This is a bit my fault, because before there were couple of 'if's and I insisted to have it shortened. After refactor the code might have lost it's readability.